fabtests/multinode/src/core.c

```diff
@@ -235,7 +286,131 @@ static int multinode_wait_for_comp()
 	return 0;
 }
 
-static inline void multinode_init_state()
+int multi_rma_write()
+{
+	int ret;
+	struct fi_msg_rma *message = (struct fi_msg_rma *) malloc(sizeof(struct fi_msg_rma));
+	struct iovec *loc_iov = (struct iovec *) malloc(sizeof(struct iovec));
+
+	message->desc = &mr_desc;
+	message->context = NULL;
+	message->rma_iov_count = 1;
+	message->iov_count = 1;
+	message->msg_iov = loc_iov;
+	message->data = 0;
+
+	while (!state.all_sends_posted) {
+
+		if (state.tx_window == 0)
+			break;
+
+		ret = pattern->next_target(&state.cur_target);
+		if (ret == -FI_ENODATA) {
+			state.all_sends_posted = true;
+			break;
+		} else if (ret < 0) {
+			return ret;
+		}
+
+		loc_iov->iov_base = (void *) (tx_buf + tx_size * state.cur_target);
+		loc_iov->iov_len = tx_size;
+
+		snprintf((char*) loc_iov->iov_base, tx_size,
+				"Hello World! from %zu to %i on the %zuth iteration, %s test",
+				pm_job.my_rank, state.cur_target, (size_t) tx_seq, pattern->name);
+
+		message->rma_iov = &pm_job.multi_iovs[state.cur_target];
+		message->addr = pm_job.fi_addrs[state.cur_target];
+		message->context = &tx_ctx_arr[state.tx_window].context;
+
+		do {
+			ret = fi_writemsg(ep, message, NO_CQ_DATA);
+		} while (ret == -FI_EAGAIN);
```

Comment:
We can't loop on EAGAIN and expect the result to change.  If we receive EAGAIN, we likely need to poll the CQ to progress the connection and free space on the send queue.